--- Python-2.7.13/Modules/_ssl.c
+++ Python-2.7.13/Modules/_ssl.c
@@ -1128,8 +1128,10 @@ _get_crl_dp(X509 *certificate) {
 #if OPENSSL_VERSION_NUMBER >= 0x10001000L
     /* Calls x509v3_cache_extensions and sets up crldp */
     X509_check_ca(certificate);
-#endif
+    dps = certificate->crldp;
+#else
     dps = X509_get_ext_d2i(certificate, NID_crl_distribution_points, NULL, NULL);
+#endif

     if (dps == NULL)
         return Py_None;
