--- libiconv-1.14/windows/windres-options.orig	2016-02-17 19:57:34.000000000 -0500
+++ libiconv-1.14/windows/windres-options	2016-02-17 19:58:01.000000000 -0500
@@ -35,6 +35,7 @@
   echo "-DPACKAGE_VERSION_MAJOR="`echo "${version}" | sed -n -e "$sed_extract_major"`
   echo "-DPACKAGE_VERSION_MINOR="`echo "${version}" | sed -n -e "$sed_extract_minor"`
   echo "-DPACKAGE_VERSION_SUBMINOR="`echo "${version}" | sed -n -e "$sed_extract_subminor"`
+  echo "${RCFLAGS}"
 } |
 {
   if test -n "$escape"; then
