--- xml_document.c.org	2023-07-04 22:56:04
+++ xml_document.c	2023-07-05 12:09:51
@@ -60,7 +60,10 @@
     rb_gc_mark(tuple->node_cache);
   }
 }
-
+#if defined(__GNUC__) && __GNUC__ >= 5
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations" // xmlDeregisterNodeDefault is deprecated as of libxml2 2.11.0
+#endif
 static void
 dealloc(void *data)
 {
@@ -74,8 +77,6 @@
 
   ruby_xfree(doc->_private);
 
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations" // xmlDeregisterNodeDefault is deprecated as of libxml2 2.11.0
   /*
    * libxml-ruby < 3.0.0 uses xmlDeregisterNodeDefault. If the user is using one of those older
    * versions, the registered callback from libxml-ruby will access the _private pointers set by
@@ -90,8 +91,9 @@
   if (xmlDeregisterNodeDefaultValue) {
     remove_private((xmlNodePtr)doc);
   }
-#pragma GCC diagnostic pop
-
+  #if defined(__GNUC__) && __GNUC__ >= 5
+  #pragma GCC diagnostic pop
+  #endif
   xmlFreeDoc(doc);
 }
 
