diff -ru nrpe-2.14-orig/src/nrpe.c nrpe-2.14/src/nrpe.c
--- nrpe-2.14-orig/src/nrpe.c	2013-04-03 18:39:30.000000000 +0200
+++ nrpe-2.14/src/nrpe.c	2013-04-03 18:41:40.000000000 +0200
@@ -348,6 +348,9 @@
 
 
 	/* else daemonize and start listening for requests... */
+#ifdef NODAEMON
+	else {
+#else
 	else if(fork()==0){
 		
 		/* we're a daemon - set up a new process group */
@@ -365,6 +368,7 @@
 
 		chdir("/");
 		/*umask(0);*/
+#endif
 
 		/* handle signals */
 		signal(SIGQUIT,sighandler);
@@ -374,6 +378,7 @@
 		/* log info to syslog facility */
 		syslog(LOG_NOTICE,"Starting up daemon");
 
+#ifndef NODAEMON
 		/* write pid file */
 		if(write_pid_file()==ERROR)
 			return STATE_CRITICAL;
@@ -383,6 +388,7 @@
 
 		/* make sure we're not root */
 		check_privileges();
+#endif
 
 		do{
 
@@ -410,8 +416,10 @@
 	
 		        }while(sigrestart==TRUE && sigshutdown==FALSE);
 
+#ifndef NODAEMON
 		/* remove pid file */
 		remove_pid_file();
+#endif
 
 		syslog(LOG_NOTICE,"Daemon shutdown\n");
 	        }
Only in nrpe-2.14/src: nrpe.c~
