diff --git a/src/server.c b/src/server.c
index 280470f6..105bcd4d 100644
--- a/src/server.c
+++ b/src/server.c
@@ -4188,6 +4188,14 @@ int main(int argc, char **argv) {
         serverLog(LL_WARNING, "Configuration loaded");
     }
 
+    if (!server.requirepass) {
+      const char *password = getenv("REDIS_PASSWORD");
+      if (password != NULL) {
+        server.requirepass = zstrdup(password);
+      }
+      unsetenv("REDIS_PASSWORD");
+    }
+
     server.supervised = redisIsSupervised(server.supervised_mode);
     int background = server.daemonize && !server.supervised;
     if (background) daemonize();
