diff --git a/src/redis.c b/src/redis.c
index 8e792c17..e31e0a28 100644
--- a/src/redis.c
+++ b/src/redis.c
@@ -3667,6 +3667,15 @@ int main(int argc, char **argv) {
     } else {
         redisLog(REDIS_WARNING, "Warning: no config file specified, using the default config. In order to specify a config file use %s /path/to/%s.conf", argv[0], server.sentinel_mode ? "sentinel" : "redis");
     }
+
+    if (!server.requirepass) {
+      const char* password = getenv("REDIS_PASSWORD");
+      if (password != NULL) {
+        server.requirepass = zstrdup(password);
+      }
+      unsetenv("REDIS_PASSWORD");
+    }
+
     if (server.daemonize) daemonize();
     initServer();
     if (server.daemonize) createPidFile();
