diff --git a/Makefile b/Makefile
index 73de5eb..4326035 100644
--- a/Makefile
+++ b/Makefile
@@ -122,7 +122,7 @@ am__append_2 = \
 #		cpio/cpio_windows.h \
 #		cpio/cpio_windows.c
 
-#am__append_9 = 
+#am__append_9 =
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps =  \
@@ -468,7 +468,7 @@ libarchive_la_OBJECTS = $(am_libarchive_la_OBJECTS)
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
 am__v_lt_0 = --silent
-am__v_lt_1 = 
+am__v_lt_1 =
 libarchive_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(libarchive_la_LDFLAGS) $(LDFLAGS) -o $@
@@ -1510,11 +1510,11 @@ am__v_P_1 = :
 AM_V_GEN = $(am__v_GEN_$(V))
 am__v_GEN_ = $(am__v_GEN_$(AM_DEFAULT_VERBOSITY))
 am__v_GEN_0 = @echo "  GEN     " $@;
-am__v_GEN_1 = 
+am__v_GEN_1 =
 AM_V_at = $(am__v_at_$(V))
 am__v_at_ = $(am__v_at_$(AM_DEFAULT_VERBOSITY))
 am__v_at_0 = @
-am__v_at_1 = 
+am__v_at_1 =
 DEFAULT_INCLUDES = -I.
 depcomp = $(SHELL) $(top_srcdir)/build/autoconf/depcomp
 am__maybe_remake_depfiles = depfiles
@@ -2225,7 +2225,7 @@ LTCOMPILE = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 AM_V_CC = $(am__v_CC_$(V))
 am__v_CC_ = $(am__v_CC_$(AM_DEFAULT_VERBOSITY))
 am__v_CC_0 = @echo "  CC      " $@;
-am__v_CC_1 = 
+am__v_CC_1 =
 CCLD = $(CC)
 LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
@@ -2233,7 +2233,7 @@ LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 AM_V_CCLD = $(am__v_CCLD_$(V))
 am__v_CCLD_ = $(am__v_CCLD_$(AM_DEFAULT_VERBOSITY))
 am__v_CCLD_0 = @echo "  CCLD    " $@;
-am__v_CCLD_1 = 
+am__v_CCLD_1 =
 SOURCES = $(libarchive_la_SOURCES) $(libarchive_fe_la_SOURCES) \
 	$(bsdcat_SOURCES) $(bsdcat_test_SOURCES) $(bsdcpio_SOURCES) \
 	$(bsdcpio_test_SOURCES) $(bsdtar_SOURCES) \
@@ -2503,23 +2503,23 @@ CC = cc -qlanglvl=extc1x
 CCDEPMODE = depmode=xlc
 CFLAGS = -g -Wall -Wformat -Wformat-security
 CPP = cc -qlanglvl=extc1x -E
-CPPFLAGS = 
+CPPFLAGS =
 CSCOPE = cscope
 CTAGS = ctags
 CYGPATH_W = echo
 DEFS = -DHAVE_CONFIG_H
 DEPDIR = .deps
 DLLTOOL = dlltool
-DSYMUTIL = 
-DUMPBIN = 
+DSYMUTIL =
+DUMPBIN =
 ECHO_C = \c
-ECHO_N = 
-ECHO_T = 
+ECHO_N =
+ECHO_T =
 EGREP = /usr/bin/grep -E
 ETAGS = etags
-EXEEXT = 
+EXEEXT =
 FGREP = /usr/bin/grep -F
-GC_SECTIONS = 
+GC_SECTIONS =
 GREP = /usr/bin/grep
 INSTALL = /opt/freeware/bin/install -c
 INSTALL_DATA = ${INSTALL} -m 644
@@ -2527,46 +2527,46 @@ INSTALL_PROGRAM = ${INSTALL}
 INSTALL_SCRIPT = ${INSTALL}
 INSTALL_STRIP_PROGRAM = $(install_sh) -c -s
 LD = /usr/bin/ld
-LDFLAGS = 
+LDFLAGS =
 LIBARCHIVE_VERSION_NUMBER = 3006002
 LIBARCHIVE_VERSION_STRING = 3.6.2
-LIBICONV = 
+LIBICONV =
 LIBOBJS =  ${LIBOBJDIR}lstat$U.o
 LIBS = -lxml2 -llzma -lbz2 -lz -lpthread
-LIBSREQUIRED = 
+LIBSREQUIRED =
 LIBTOOL = $(SHELL) $(top_builddir)/libtool
-LIBXML2_PC_CFLAGS = 
-LIBXML2_PC_LIBS = 
-LIPO = 
+LIBXML2_PC_CFLAGS =
+LIBXML2_PC_LIBS =
+LIPO =
 LN_S = ln -s
-LTLIBICONV = 
+LTLIBICONV =
 LTLIBOBJS =  ${LIBOBJDIR}lstat$U.lo
-LT_SYS_LIBRARY_PATH = 
-MAINT = 
+LT_SYS_LIBRARY_PATH =
+MAINT =
 MAKEINFO = ${SHELL} '/opt/libarchive-3.5.0/libarchive-3.6.2/build/autoconf/missing' makeinfo
 MANIFEST_TOOL = :
 MKDIR_P = /opt/freeware/bin/mkdir -p
 NM = /usr/bin/nm -B
-NMEDIT = 
+NMEDIT =
 OBJDUMP = objdump
 OBJEXT = o
-OTOOL = 
-OTOOL64 = 
+OTOOL =
+OTOOL64 =
 PACKAGE = libarchive
 PACKAGE_BUGREPORT = libarchive-discuss@googlegroups.com
 PACKAGE_NAME = libarchive
 PACKAGE_STRING = libarchive 3.6.2
 PACKAGE_TARNAME = libarchive
-PACKAGE_URL = 
+PACKAGE_URL =
 PACKAGE_VERSION = 3.6.2
 PATH_SEPARATOR = :
-PKG_CONFIG = 
-PKG_CONFIG_LIBDIR = 
-PKG_CONFIG_PATH = 
-PLATFORMCPPFLAGS = 
+PKG_CONFIG =
+PKG_CONFIG_LIBDIR =
+PKG_CONFIG_PATH =
+PLATFORMCPPFLAGS =
 RANLIB = ranlib
 SED = /usr/bin/sed
-SET_MAKE = 
+SET_MAKE =
 SHELL = /bin/sh
 STRIP = strip
 VERSION = 3.6.2
@@ -2575,16 +2575,16 @@ abs_srcdir = /opt/libarchive-3.5.0/libarchive-3.6.2
 abs_top_builddir = /opt/libarchive-3.5.0/libarchive-3.6.2
 abs_top_srcdir = /opt/libarchive-3.5.0/libarchive-3.6.2
 ac_ct_AR = ar
-ac_ct_CC = 
-ac_ct_DUMPBIN = 
+ac_ct_CC =
+ac_ct_DUMPBIN =
 am__include = include
 am__leading_dot = .
-am__quote = 
+am__quote =
 am__tar = $${TAR-tar} chof - "$$tardir"
 am__untar = $${TAR-tar} xf -
 bindir = ${exec_prefix}/bin
 build = powerpc-ibm-aix7.1.5.0
-build_alias = 
+build_alias =
 build_cpu = powerpc
 build_os = aix7.1.5.0
 build_vendor = ibm
@@ -2595,7 +2595,7 @@ docdir = ${datarootdir}/doc/${PACKAGE_TARNAME}
 dvidir = ${docdir}
 exec_prefix = ${prefix}
 host = powerpc-ibm-aix7.1.5.0
-host_alias = 
+host_alias =
 host_cpu = powerpc
 host_os = aix7.1.5.0
 host_vendor = ibm
@@ -2619,8 +2619,8 @@ sbindir = ${exec_prefix}/sbin
 sharedstatedir = ${prefix}/com
 srcdir = .
 sysconfdir = ${prefix}/etc
-target_alias = 
-top_build_prefix = 
+target_alias =
+top_build_prefix =
 top_builddir = .
 top_srcdir = .
 AUTOMAKE_OPTIONS = foreign subdir-objects
@@ -3541,9 +3541,9 @@ bsdtar_SOURCES = tar/bsdtar.c tar/bsdtar.h tar/bsdtar_platform.h \
 	tar/cmdline.c tar/creation_set.c tar/read.c tar/subst.c \
 	tar/util.c tar/write.c $(am__append_7)
 bsdtar_DEPENDENCIES = libarchive.la libarchive_fe.la
-bsdtar_ldstatic = 
+bsdtar_ldstatic =
 #bsdtar_ldstatic = -static
-bsdtar_ccstatic = 
+bsdtar_ccstatic =
 #bsdtar_ccstatic = -DLIBARCHIVE_STATIC
 bsdtar_LDADD = libarchive.la libarchive_fe.la $(LTLIBICONV)
 bsdtar_CPPFLAGS = -I$(top_srcdir)/libarchive -I$(top_srcdir)/libarchive_fe $(bsdtar_ccstatic) $(PLATFORMCPPFLAGS)
@@ -3555,9 +3555,9 @@ bsdtar_EXTRA_DIST = \
 	tar/CMakeLists.txt \
 	tar/config_freebsd.h
 
-bsdtar_man_MANS = 
+bsdtar_man_MANS =
 #bsdtar_man_MANS = tar/bsdtar.1
-bsdtar_programs = 
+bsdtar_programs =
 #bsdtar_programs = bsdtar
 
 #
@@ -3640,9 +3640,9 @@ bsdtar_test_CPPFLAGS = \
 	-I$(top_builddir)/tar/test \
 	$(PLATFORMCPPFLAGS)
 
-bsdtar_test_programs = 
+bsdtar_test_programs =
 #bsdtar_test_programs = bsdtar_test
-bsdtar_TESTS_ENVIRONMENT = 
+bsdtar_TESTS_ENVIRONMENT =
 #bsdtar_TESTS_ENVIRONMENT = BSDTAR=`cd $(top_builddir);/bin/pwd`/bsdtar$(EXEEXT) BSDTAR_TEST_FILES=`cd $(top_srcdir);/bin/pwd`/tar/test
 bsdtar_test_EXTRA_DIST = \
 	tar/test/list.h \
@@ -3676,9 +3676,9 @@ bsdtar_test_EXTRA_DIST = \
 bsdcpio_SOURCES = cpio/cmdline.c cpio/cpio.c cpio/cpio.h \
 	cpio/cpio_platform.h $(am__append_8)
 bsdcpio_DEPENDENCIES = libarchive.la libarchive_fe.la
-bsdcpio_ldstatic = 
+bsdcpio_ldstatic =
 #bsdcpio_ldstatic = -static
-bsdcpio_ccstatic = 
+bsdcpio_ccstatic =
 #bsdcpio_ccstatic = -DLIBARCHIVE_STATIC
 bsdcpio_LDADD = libarchive_fe.la libarchive.la $(LTLIBICONV)
 bsdcpio_CPPFLAGS = -I$(top_srcdir)/libarchive -I$(top_srcdir)/libarchive_fe $(bsdcpio_ccstatic) $(PLATFORMCPPFLAGS)
@@ -3690,11 +3690,11 @@ bsdcpio_EXTRA_DIST = \
 	cpio/CMakeLists.txt \
 	cpio/config_freebsd.h
 
-bsdcpio_man_MANS = 
+bsdcpio_man_MANS =
 
 # Manpages to install
 #bsdcpio_man_MANS = cpio/bsdcpio.1
-bsdcpio_programs = 
+bsdcpio_programs =
 #bsdcpio_programs = bsdcpio
 
 #
@@ -3761,9 +3761,9 @@ bsdcpio_test_CPPFLAGS = \
 	$(PLATFORMCPPFLAGS)
 
 bsdcpio_test_LDADD = libarchive_fe.la
-bsdcpio_test_programs = 
+bsdcpio_test_programs =
 #bsdcpio_test_programs = bsdcpio_test
-bsdcpio_TESTS_ENVIRONMENT = 
+bsdcpio_TESTS_ENVIRONMENT =
 #bsdcpio_TESTS_ENVIRONMENT = BSDCPIO=`cd $(top_builddir);/bin/pwd`/bsdcpio$(EXEEXT) BSDCPIO_TEST_FILES=`cd $(top_srcdir);/bin/pwd`/cpio/test
 bsdcpio_test_EXTRA_DIST = \
 	cpio/test/list.h \
@@ -3803,9 +3803,9 @@ bsdcpio_test_EXTRA_DIST = \
 bsdcat_SOURCES = cat/bsdcat.c cat/bsdcat.h cat/bsdcat_platform.h \
 	cat/cmdline.c $(am__append_9)
 bsdcat_DEPENDENCIES = libarchive.la libarchive_fe.la
-bsdcat_ldstatic = 
+bsdcat_ldstatic =
 #bsdcat_ldstatic = -static
-bsdcat_ccstatic = 
+bsdcat_ccstatic =
 #bsdcat_ccstatic = -DLIBARCHIVE_STATIC
 bsdcat_LDADD = libarchive_fe.la libarchive.la $(LTLIBICONV)
 bsdcat_CPPFLAGS = -I$(top_srcdir)/libarchive -I$(top_srcdir)/libarchive_fe $(bsdcat_ccstatic) $(PLATFORMCPPFLAGS)
@@ -3814,11 +3814,11 @@ bsdcat_EXTRA_DIST = \
 	cat/bsdcat.1 \
 	cat/CMakeLists.txt
 
-bsdcat_man_MANS = 
+bsdcat_man_MANS =
 
 # Manpages to install
 #bsdcat_man_MANS = cat/bsdcat.1
-bsdcat_programs = 
+bsdcat_programs =
 #bsdcat_programs = bsdcat
 
 #
@@ -3854,9 +3854,9 @@ bsdcat_test_CPPFLAGS = \
 	$(PLATFORMCPPFLAGS)
 
 bsdcat_test_LDADD = libarchive_fe.la
-bsdcat_test_programs = 
+bsdcat_test_programs =
 #bsdcat_test_programs = bsdcat_test
-bsdcat_TESTS_ENVIRONMENT = 
+bsdcat_TESTS_ENVIRONMENT =
 #bsdcat_TESTS_ENVIRONMENT = BSDCAT=`cd $(top_builddir);/bin/pwd`/bsdcat$(EXEEXT) BSDCAT_TEST_FILES=`cd $(top_srcdir);/bin/pwd`/cat/test
 bsdcat_test_EXTRA_DIST = \
 	cat/test/list.h \
@@ -3919,7 +3919,7 @@ config.h: stamp-h1
 stamp-h1: $(srcdir)/config.h.in $(top_builddir)/config.status
 	@rm -f stamp-h1
 	cd $(top_builddir) && $(SHELL) ./config.status config.h
-$(srcdir)/config.h.in:  $(am__configure_deps) 
+$(srcdir)/config.h.in:  $(am__configure_deps)
 	($(am__cd) $(top_srcdir) && $(AUTOHEADER))
 	rm -f stamp-h1
 	touch $@
@@ -4001,6 +4001,7 @@ install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	  echo " $(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=install $(INSTALL) $(INSTALL_STRIP_FLAG) $$list2 '$(DESTDIR)$(libdir)'"; \
 	  $(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=install $(INSTALL) $(INSTALL_STRIP_FLAG) $$list2 "$(DESTDIR)$(libdir)"; \
 	}
+	$(LN_S) -f $(libdir)/libarchive.so.13 $(libdir)/libarchive.so
 
 uninstall-libLTLIBRARIES:
 	@$(NORMAL_UNINSTALL)
@@ -4368,7 +4369,7 @@ libarchive/archive_disk_acl_darwin.lo: libarchive/$(am__dirstamp) \
 libarchive/archive_disk_acl_freebsd.lo: libarchive/$(am__dirstamp) \
 	libarchive/$(DEPDIR)/$(am__dirstamp)
 
-libarchive.la: $(libarchive_la_OBJECTS) $(libarchive_la_DEPENDENCIES) $(EXTRA_libarchive_la_DEPENDENCIES) 
+libarchive.la: $(libarchive_la_OBJECTS) $(libarchive_la_DEPENDENCIES) $(EXTRA_libarchive_la_DEPENDENCIES)
 	$(AM_V_CCLD)$(libarchive_la_LINK) -rpath $(libdir) $(libarchive_la_OBJECTS) $(libarchive_la_LIBADD) $(LIBS)
 libarchive_fe/$(am__dirstamp):
 	@$(MKDIR_P) libarchive_fe
@@ -4383,7 +4384,7 @@ libarchive_fe/la-line_reader.lo: libarchive_fe/$(am__dirstamp) \
 libarchive_fe/la-passphrase.lo: libarchive_fe/$(am__dirstamp) \
 	libarchive_fe/$(DEPDIR)/$(am__dirstamp)
 
-libarchive_fe.la: $(libarchive_fe_la_OBJECTS) $(libarchive_fe_la_DEPENDENCIES) $(EXTRA_libarchive_fe_la_DEPENDENCIES) 
+libarchive_fe.la: $(libarchive_fe_la_OBJECTS) $(libarchive_fe_la_DEPENDENCIES) $(EXTRA_libarchive_fe_la_DEPENDENCIES)
 	$(AM_V_CCLD)$(LINK)  $(libarchive_fe_la_OBJECTS) $(libarchive_fe_la_LIBADD) $(LIBS)
 cat/$(am__dirstamp):
 	@$(MKDIR_P) cat
@@ -4396,7 +4397,7 @@ cat/bsdcat-bsdcat.$(OBJEXT): cat/$(am__dirstamp) \
 cat/bsdcat-cmdline.$(OBJEXT): cat/$(am__dirstamp) \
 	cat/$(DEPDIR)/$(am__dirstamp)
 
-bsdcat$(EXEEXT): $(bsdcat_OBJECTS) $(bsdcat_DEPENDENCIES) $(EXTRA_bsdcat_DEPENDENCIES) 
+bsdcat$(EXEEXT): $(bsdcat_OBJECTS) $(bsdcat_DEPENDENCIES) $(EXTRA_bsdcat_DEPENDENCIES)
 	@rm -f bsdcat$(EXEEXT)
 	$(AM_V_CCLD)$(bsdcat_LINK) $(bsdcat_OBJECTS) $(bsdcat_LDADD) $(LIBS)
 test_utils/$(am__dirstamp):
@@ -4454,7 +4455,7 @@ cat/test/bsdcat_test-test_stdin.$(OBJEXT): cat/test/$(am__dirstamp) \
 cat/test/bsdcat_test-test_version.$(OBJEXT): cat/test/$(am__dirstamp) \
 	cat/test/$(DEPDIR)/$(am__dirstamp)
 
-bsdcat_test$(EXEEXT): $(bsdcat_test_OBJECTS) $(bsdcat_test_DEPENDENCIES) $(EXTRA_bsdcat_test_DEPENDENCIES) 
+bsdcat_test$(EXEEXT): $(bsdcat_test_OBJECTS) $(bsdcat_test_DEPENDENCIES) $(EXTRA_bsdcat_test_DEPENDENCIES)
 	@rm -f bsdcat_test$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(bsdcat_test_OBJECTS) $(bsdcat_test_LDADD) $(LIBS)
 cpio/$(am__dirstamp):
@@ -4470,7 +4471,7 @@ cpio/bsdcpio-cpio.$(OBJEXT): cpio/$(am__dirstamp) \
 cpio/bsdcpio-cpio_windows.$(OBJEXT): cpio/$(am__dirstamp) \
 	cpio/$(DEPDIR)/$(am__dirstamp)
 
-bsdcpio$(EXEEXT): $(bsdcpio_OBJECTS) $(bsdcpio_DEPENDENCIES) $(EXTRA_bsdcpio_DEPENDENCIES) 
+bsdcpio$(EXEEXT): $(bsdcpio_OBJECTS) $(bsdcpio_DEPENDENCIES) $(EXTRA_bsdcpio_DEPENDENCIES)
 	@rm -f bsdcpio$(EXEEXT)
 	$(AM_V_CCLD)$(bsdcpio_LINK) $(bsdcpio_OBJECTS) $(bsdcpio_LDADD) $(LIBS)
 test_utils/bsdcpio_test-test_utils.$(OBJEXT):  \
@@ -4584,7 +4585,7 @@ cpio/test/bsdcpio_test-test_passthrough_dotdot.$(OBJEXT):  \
 cpio/test/bsdcpio_test-test_passthrough_reverse.$(OBJEXT):  \
 	cpio/test/$(am__dirstamp) cpio/test/$(DEPDIR)/$(am__dirstamp)
 
-bsdcpio_test$(EXEEXT): $(bsdcpio_test_OBJECTS) $(bsdcpio_test_DEPENDENCIES) $(EXTRA_bsdcpio_test_DEPENDENCIES) 
+bsdcpio_test$(EXEEXT): $(bsdcpio_test_OBJECTS) $(bsdcpio_test_DEPENDENCIES) $(EXTRA_bsdcpio_test_DEPENDENCIES)
 	@rm -f bsdcpio_test$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(bsdcpio_test_OBJECTS) $(bsdcpio_test_LDADD) $(LIBS)
 tar/$(am__dirstamp):
@@ -4610,7 +4611,7 @@ tar/bsdtar-write.$(OBJEXT): tar/$(am__dirstamp) \
 tar/bsdtar-bsdtar_windows.$(OBJEXT): tar/$(am__dirstamp) \
 	tar/$(DEPDIR)/$(am__dirstamp)
 
-bsdtar$(EXEEXT): $(bsdtar_OBJECTS) $(bsdtar_DEPENDENCIES) $(EXTRA_bsdtar_DEPENDENCIES) 
+bsdtar$(EXEEXT): $(bsdtar_OBJECTS) $(bsdtar_DEPENDENCIES) $(EXTRA_bsdtar_DEPENDENCIES)
 	@rm -f bsdtar$(EXEEXT)
 	$(AM_V_CCLD)$(bsdtar_LINK) $(bsdtar_OBJECTS) $(bsdtar_LDADD) $(LIBS)
 test_utils/bsdtar_test-test_utils.$(OBJEXT):  \
@@ -4758,7 +4759,7 @@ tar/test/bsdtar_test-test_version.$(OBJEXT): tar/test/$(am__dirstamp) \
 tar/test/bsdtar_test-test_windows.$(OBJEXT): tar/test/$(am__dirstamp) \
 	tar/test/$(DEPDIR)/$(am__dirstamp)
 
-bsdtar_test$(EXEEXT): $(bsdtar_test_OBJECTS) $(bsdtar_test_DEPENDENCIES) $(EXTRA_bsdtar_test_DEPENDENCIES) 
+bsdtar_test$(EXEEXT): $(bsdtar_test_OBJECTS) $(bsdtar_test_DEPENDENCIES) $(EXTRA_bsdtar_test_DEPENDENCIES)
 	@rm -f bsdtar_test$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(bsdtar_test_OBJECTS) $(bsdtar_test_LDADD) $(LIBS)
 libarchive/test-archive_acl.$(OBJEXT): libarchive/$(am__dirstamp) \
@@ -5987,7 +5988,7 @@ libarchive/test/test-test_zip_filename_encoding.$(OBJEXT):  \
 	libarchive/test/$(am__dirstamp) \
 	libarchive/test/$(DEPDIR)/$(am__dirstamp)
 
-libarchive_test$(EXEEXT): $(libarchive_test_OBJECTS) $(libarchive_test_DEPENDENCIES) $(EXTRA_libarchive_test_DEPENDENCIES) 
+libarchive_test$(EXEEXT): $(libarchive_test_OBJECTS) $(libarchive_test_DEPENDENCIES) $(EXTRA_libarchive_test_DEPENDENCIES)
 	@rm -f libarchive_test$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(libarchive_test_OBJECTS) $(libarchive_test_LDADD) $(LIBS)
 
