--- libarchive/archive_write_disk_posix.c.org	2025-08-01 16:13:09
+++ libarchive/archive_write_disk_posix.c	2025-08-01 16:50:50
@@ -229,11 +229,11 @@
 	int64_t			 skip_file_ino;
 	time_t			 start_time;
 
-	int64_t (*lookup_gid)(void *private, const char *gname, int64_t gid);
-	void  (*cleanup_gid)(void *private);
+	int64_t (*lookup_gid)(const void *private, const char *gname, int64_t gid);
+	void  (*cleanup_gid)(const void *private);
 	void			*lookup_gid_data;
-	int64_t (*lookup_uid)(void *private, const char *uname, int64_t uid);
-	void  (*cleanup_uid)(void *private);
+	int64_t (*lookup_uid)(const void *private, const char *uname, int64_t uid);
+	void  (*cleanup_uid)(const void *private);
 	void			*lookup_uid_data;
 
 	/*
@@ -1464,7 +1464,7 @@
 	int ret;
 
 	if (a->decmpfs_block_count == (unsigned)-1) {
-		void *new_block;
+		const void *new_block;
 		size_t new_size;
 		unsigned int block_count;
 
@@ -1927,9 +1927,9 @@
 
 int
 archive_write_disk_set_group_lookup(struct archive *_a,
-    void *private_data,
-    la_int64_t (*lookup_gid)(void *private, const char *gname, la_int64_t gid),
-    void (*cleanup_gid)(void *private))
+    const void *private_data,
+    la_int64_t (*lookup_gid)(const void *private, const char *gname, la_int64_t gid),
+    void (*cleanup_gid)(const void *private))
 {
 	struct archive_write_disk *a = (struct archive_write_disk *)_a;
 	archive_check_magic(&a->archive, ARCHIVE_WRITE_DISK_MAGIC,
@@ -1946,9 +1946,9 @@
 
 int
 archive_write_disk_set_user_lookup(struct archive *_a,
-    void *private_data,
-    int64_t (*lookup_uid)(void *private, const char *uname, int64_t uid),
-    void (*cleanup_uid)(void *private))
+    const void *private_data,
+    int64_t (*lookup_uid)(const void *private, const char *uname, int64_t uid),
+    void (*cleanup_uid)(const void *private))
 {
 	struct archive_write_disk *a = (struct archive_write_disk *)_a;
 	archive_check_magic(&a->archive, ARCHIVE_WRITE_DISK_MAGIC,
