--- a/tests/test_security_curve.cpp
+++ b/tests/test_security_curve.cpp
@@ -235,7 +235,7 @@ int main (void)

     ip4addr.sin_family = AF_INET;
     ip4addr.sin_port = htons (9998);
-    inet_pton (AF_INET, "127.0.0.1", &ip4addr.sin_addr);
+    ip4addr.sin_addr.s_addr = inet_addr ("127.0.0.1");

     s = socket (AF_INET, SOCK_STREAM, IPPROTO_TCP);
     rc = connect (s, (struct sockaddr*) &ip4addr, sizeof (ip4addr));
diff --git a/tests/test_security_null.cpp b/tests/test_security_null.cpp
index 6b74e8c..9b87aa3 100644
--- a/tests/test_security_null.cpp
+++ b/tests/test_security_null.cpp
@@ -148,7 +148,7 @@ int main (void)

     ip4addr.sin_family = AF_INET;
     ip4addr.sin_port = htons(9003);
-    inet_pton(AF_INET, "127.0.0.1", &ip4addr.sin_addr);
+    ip4addr.sin_addr.s_addr = inet_addr ("127.0.0.1");

     s = socket (AF_INET, SOCK_STREAM, IPPROTO_TCP);
     rc = connect (s, (struct sockaddr*) &ip4addr, sizeof ip4addr);
diff --git a/tests/test_security_plain.cpp b/tests/test_security_plain.cpp
index c257840..cd66790 100644
--- a/tests/test_security_plain.cpp
+++ b/tests/test_security_plain.cpp
@@ -154,7 +154,7 @@ int main (void)

     ip4addr.sin_family = AF_INET;
     ip4addr.sin_port = htons (9998);
-    inet_pton (AF_INET, "127.0.0.1", &ip4addr.sin_addr);
+    ip4addr.sin_addr.s_addr = inet_addr ("127.0.0.1");

     s = socket (AF_INET, SOCK_STREAM, IPPROTO_TCP);
     rc = connect (s, (struct sockaddr*) &ip4addr, sizeof (ip4addr));
