--- configure.in.orig	2012-10-12 07:24:42.000000000 +0000
+++ configure.in	2013-02-11 04:53:25.988951882 +0000
@@ -1919,11 +1919,11 @@
 
 AS_CASE(["$target_os"],
 [freebsd*], [
-    AC_CHECK_HEADERS([/usr/local/include/execinfo.h])
-    if test "x$ac_cv_header__usr_local_include_execinfo_h" = xyes; then :
-	RUBY_APPEND_OPTION(CPPFLAGS, -I/usr/local/include)
-	LDFLAGS="${LDFLAGS:+$LDFLAGS }-L/usr/local/lib"
-	DLDFLAGS="${DLDFLAGS:+$DLDFLAGS }-L/usr/local/lib"
+    AC_CHECK_HEADERS([/opt/chef/embedded/include/execinfo.h])
+    if test "xyes" = xyes; then :
+	RUBY_APPEND_OPTION(CPPFLAGS, -I/opt/chef/embedded/include)
+	LDFLAGS="${LDFLAGS:+$LDFLAGS }-L/opt/chef/embedded/lib"
+	DLDFLAGS="${DLDFLAGS:+$DLDFLAGS }-L/opt/chef/embedded/lib"
 	AC_CHECK_LIB([execinfo], [backtrace])
     fi])
 AC_CHECK_FUNCS(backtrace)
