diff --git a/configure.in b/configure.in
index 29b63c2..91f42df 100644
--- a/configure.in
+++ b/configure.in
@@ -1189,6 +1189,7 @@ main()
 		{ test "$target_cpu" = x64 && ac_cv_func___builtin_setjmp=no; }
 		AC_CHECK_TYPE([NET_LUID], [], [],
 			      [@%:@include <winsock2.h>
+			      @%:@include <windows.h>
 			      @%:@include <iphlpapi.h>])
 		if test x"$ac_cv_type_NET_LUID" = xyes; then
 		    AC_DEFINE(HAVE_TYPE_NET_LUID, 1)
@@ -1500,7 +1501,7 @@ AC_CACHE_CHECK([for printf prefix for $1], [rb_cv_pri_prefix_]AS_TR_SH($1),[
 	    @%:@include <stddef.h>
             @%:@ifdef __GNUC__
             @%:@define PRINTF_ARGS(decl, string_index, first_to_check) \
-              decl __attribute__((format(printf, string_index, first_to_check)))
+              decl __attribute__((format(gnu_printf, string_index, first_to_check)))
             @%:@else
             @%:@define PRINTF_ARGS(decl, string_index, first_to_check) decl
             @%:@endif
diff --git a/include/ruby/win32.h b/include/ruby/win32.h
index 47f68b8..2bdf3c2 100644
--- a/include/ruby/win32.h
+++ b/include/ruby/win32.h
@@ -715,8 +715,10 @@ struct tms {
 
 int rb_w32_times(struct tms *);
 
+#ifndef __MINGW64_VERSION_MAJOR
 struct tm *gmtime_r(const time_t *, struct tm *);
 struct tm *localtime_r(const time_t *, struct tm *);
+#endif
 
 /* thread stuff */
 int  rb_w32_sleep(unsigned long msec);
diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index 4614c35..48eebc0 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -276,14 +276,12 @@ def open_for_install(path, mode)
 
 def with_destdir(dir)
   return dir if !$destdir or $destdir.empty?
-  dir = dir.sub(/\A\w:/, '') if File::PATH_SEPARATOR == ';'
   $destdir + dir
 end
 
 def without_destdir(dir)
-  return dir if !$destdir or $destdir.empty? or !dir.start_with?($destdir)
-  dir = dir.sub(/\A\w:/, '') if File::PATH_SEPARATOR == ';'
-  dir[$destdir.size..-1]
+  return dir if !$destdir or $destdir.empty?
+  dir.start_with?($destdir) ? dir[$destdir.size..-1] : dir
 end
 
 def prepare(mesg, basedir, subdirs=nil)
diff --git a/win32/win32.c b/win32/win32.c
index 47408e0..d095df5 100644
--- a/win32/win32.c
+++ b/win32/win32.c
@@ -7562,6 +7562,7 @@ systemtime_to_localtime(TIME_ZONE_INFORMATION *tz, SYSTEMTIME *gst, SYSTEMTIME *
 # define localtime_s _localtime64_s
 #endif
 
+#ifndef __MINGW64_VERSION_MAJOR
 /* License: Ruby's */
 struct tm *
 gmtime_r(const time_t *tp, struct tm *rp)
@@ -7609,6 +7610,7 @@ localtime_r(const time_t *tp, struct tm *rp)
 #endif
     return rp;
 }
+#endif
 
 /* License: Ruby's */
 int
