diff -ur a/template/Makefile.in b/template/Makefile.in
--- a/template/Makefile.in	2021-04-07 20:49:02.516848080 -0700
+++ b/template/Makefile.in	2021-04-07 20:49:57.504305399 -0700
@@ -75,7 +75,7 @@
 CC_VERSION = @CC_VERSION@
 OUTFLAG = @OUTFLAG@$(empty)
 COUTFLAG = @COUTFLAG@$(empty)
-CPPOUTFLAG = $(COUTFLAG)
+CPPOUTFLAG = >
 ARCH_FLAG = @ARCH_FLAG@
 CFLAGS_NO_ARCH = @CFLAGS@
 CFLAGS = $(CFLAGS_NO_ARCH) $(ARCH_FLAG)
