--- src/tar.c   2016-03-23 22:42:14.000000000 -0700
+++ src/tar.c.new       2016-06-15 15:04:59.000000000 -0700
@@ -547,11 +547,7 @@
   {"no-delay-directory-restore", NO_DELAY_DIRECTORY_RESTORE_OPTION, 0, 0,
    N_("cancel the effect of --delay-directory-restore option"), GRID+1 },
   {"sort", SORT_OPTION, N_("ORDER"), 0,
-#if D_INO_IN_DIRENT
-   N_("directory sorting order: none (default), name or inode"
-#else
    N_("directory sorting order: none (default) or name"
-#endif
      ), GRID+1 },
 #undef GRID
