*** src/unlink.c        Thu Apr  5 18:33:48 2018
--- src/unlink.c.new    Thu Apr  5 18:33:48 2018
***************
*** 124,132 ****
                    case ENOENT:
                      /* nothing to worry about */
                      break;
-                   case EEXIST:
-                     /* OpenSolaris >=10 sets EEXIST instead of ENOTEMPTY
-                        if trying to remove a non-empty directory */
                    case ENOTEMPTY:
                      /* Keep the record in list, in the hope we'll
                         be able to remove it later */
