diff --git a/pcap/pcap.go b/pcap/pcap.go
index 30c024a..24193a1 100644
--- a/pcap/pcap.go
+++ b/pcap/pcap.go
@@ -9,12 +9,13 @@ package pcap

 /*
 #cgo solaris LDFLAGS: -L /opt/local/lib -lpcap
-#cgo linux LDFLAGS: -lpcap
+#cgo linux CFLAGS: -I/opt/datadog-agent/embedded/include
+#cgo linux LDFLAGS: /opt/datadog-agent/embedded/lib/libpcap.a
 #cgo dragonfly LDFLAGS: -lpcap
 #cgo freebsd LDFLAGS: -lpcap
 #cgo openbsd LDFLAGS: -lpcap
 #cgo netbsd LDFLAGS: -lpcap
-#cgo darwin LDFLAGS: -lpcap
+#cgo darwin LDFLAGS: /usr/local/Cellar/libpcap/1.7.4/lib/libpcap.a
 #cgo windows CFLAGS: -I C:/WpdPack/Include
 #cgo windows,386 LDFLAGS: -L C:/WpdPack/Lib -lwpcap
 #cgo windows,amd64 LDFLAGS: -L C:/WpdPack/Lib/x64 -lwpcap
