--- src/config.h	2025-10-21 21:35:42
+++ src/config.h.mod	2025-11-20 23:35:15
@@ -216,7 +216,14 @@
 #include <sys/types.h> /* This will likely define BYTE_ORDER */
 
 #ifndef BYTE_ORDER
-#if (BSD >= 199103)
+#ifndef BSD
+#define BSD 0
+#endif
+
+#ifndef __has_attribute
+#define __has_attribute(x) 0
+#endif
+
 #include <machine/endian.h>
 #else
 #if defined(linux) || defined(__linux__)
@@ -375,7 +382,7 @@
 #endif
 
 /* Check if we can compile x86 SIMD code */
-#if defined(__x86_64__) && ((defined(__GNUC__) && __GNUC__ >= 5) || (defined(__clang__) && __clang_major__ >= 4)) && defined(__has_attribute) && __has_attribute(target)
+#if defined(__x86_64__) && ((defined(__GNUC__) && __GNUC__ >= 5) || (defined(__clang__) && __clang_major__ >= 4)) && __has_attribute(target)
 #define HAVE_X86_SIMD 1
 #else
 #define HAVE_X86_SIMD 0
