diff --git a/Makefile.shared b/Makefile.shared
index e8d222a..02ec85b 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -283,13 +283,12 @@ link_a.cygwin:
 	base=-Wl,--enable-auto-image-base; \
 	if expr $(PLATFORM) : 'mingw' > /dev/null; then \
 		case $(LIBNAME) in \
 		esac; \
 		SHLIB_SOVER=32; \
 		extras="$(LIBNAME).def"; \
 		$(PERL) util/mkdef.pl 32 $$SHLIB > $$extras; \
 	fi; \
 	dll_name=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX; \
 	$(PERL) util/mkrc.pl $$dll_name | \
