diff --git a/Makefile.shared b/Makefile.shared
index e753f44..fb88110 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -261,7 +261,7 @@ link_o.cygwin:
 	SHLIB=cyg$(LIBNAME); \
 	base=-Wl,--enable-auto-image-base; \
 	deffile=; \
-	if expr $(PLATFORM) : 'mingw' > /dev/null; then \
+	if expr $(PLATFORM) : 'mingw64' > /dev/null; then \
 		SHLIB=$(LIBNAME)eay32; base=; \
 		if test -f $(LIBNAME)eay32.def; then \
 			deffile=$(LIBNAME)eay32.def; \
@@ -281,10 +281,10 @@ link_a.cygwin:
 	SHLIB=cyg$(LIBNAME); SHLIB_SOVER=-$(LIBVERSION); SHLIB_SUFFIX=.dll; \
 	dll_name=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX; extras=; \
 	base=-Wl,--enable-auto-image-base; \
-	if expr $(PLATFORM) : 'mingw' > /dev/null; then \
+	if expr $(PLATFORM) : 'mingw64' > /dev/null; then \
 		case $(LIBNAME) in \
-			crypto) SHLIB=libeay;; \
-			ssl) SHLIB=ssleay;; \
+			crypto) SHLIB=libeay; base=-Wl,--disable-dynamicbase,--image-base,0x64000000;; \
+			ssl) SHLIB=ssleay; base=-Wl,--disable-dynamicbase,--image-base,0x65000000;; \
 		esac; \
 		SHLIB_SOVER=32; \
 		extras="$(LIBNAME).def"; \
