diff --git a/Makefile.shared b/Makefile.shared
index e8d222a..ff854bf 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -7,6 +7,7 @@
 
 # CC contains the current compiler.  This one MUST be defined
 CC=cc
+# CFLAGS=$(CFLAG) -DOPENSSL_FIPS_DEBUGGER
 CFLAGS=$(CFLAG)
 # LDFLAGS contains flags to be used when temporary object files (when building
 # shared libraries) are created, or when an application is linked.
@@ -267,6 +268,12 @@ link_o.cygwin:
 			deffile=$(LIBNAME)eay32.def; \
 		fi; \
 	fi; \
+	if expr $(PLATFORM) : 'mingw64' > /dev/null; then \
+		SHLIB=$(LIBNAME)eay32; base=; \
+		if test -f $(LIBNAME)eay32.def; then \
+			deffile=$(LIBNAME)eay32.def; \
+		fi; \
+	fi; \
 	SHLIB_SUFFIX=.dll; \
 	LIBVERSION="$(LIBVERSION)"; \
 	SHLIB_SOVER=${LIBVERSION:+"-$(LIBVERSION)"}; \
@@ -281,15 +288,16 @@ link_a.cygwin:
 	SHLIB=cyg$(LIBNAME); SHLIB_SOVER=-$(LIBVERSION); SHLIB_SUFFIX=.dll; \
 	dll_name=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX; extras=; \
 	base=-Wl,--enable-auto-image-base; \
+	echo "DEBUG <><><><> $(PLATFORM) $(LIBNAME) <><><><>"; \
 	if expr $(PLATFORM) : 'mingw' > /dev/null; then \
+		echo "DEBUG <><><><> IN MINGW <><><><> "; \
 		case $(LIBNAME) in \
-			crypto) SHLIB=libeay;; \
-			ssl) SHLIB=ssleay;; \
+			crypto) SHLIB=libeay; base=-Wl,--image-base,0x64000000;; \
+			ssl) SHLIB=ssleay; base=-Wl,--image-base,0x65000000;; \
 		esac; \
 		SHLIB_SOVER=32; \
 		extras="$(LIBNAME).def"; \
 		$(PERL) util/mkdef.pl 32 $$SHLIB > $$extras; \
-		base=; [ $(LIBNAME) = "crypto" -a -n "$(FIPSCANLIB)" ] && base=-Wl,--image-base,0x63000000; \
 	fi; \
 	dll_name=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX; \
 	$(PERL) util/mkrc.pl $$dll_name | \
