diff --git a/python/Makefile.am b/python/Makefile.am
index 2d2edd5..a0d1351 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -33,7 +33,7 @@ all-local: nghttp2.c
 	$(PYTHON) setup.py build
 
 install-exec-local:
-	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
+	$(PYTHON) setup.py easy_install -Z --prefix=$(DESTDIR)$(prefix) .
 
 uninstall-local:
 	rm -f $(DESTDIR)$(libdir)/python*/site-packages/nghttp2.so
